daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -126,6 +126,11 @@ private[lf] object PartialTransaction {
     keys = Map.empty,
   )
 
+  sealed abstract class Result extends Product with Serializable
+  final case class CompleteTransaction(tx: SubmittedTransaction) extends Result
+  final case class IncompleteTransaction(ptx: PartialTransaction) extends Result
+  final case class SerializationError(msg: String) extends Result
```

Comment:
If you make these two extend `RuntimeException` (and make `Result` a trait), you can just throw them directly.

```suggestion
  final case class IncompleteTransaction(ptx: PartialTransaction)
    extends RuntimeException(s"Unexpected abort: $ptx")
    with Result
  final case class SerializationError(msg: String)
    extends RuntimeException(s"Cannot serialize the transaction: $msg")
    with Result
```

Reply:
I definitively do not want to throw any of this. The important of `PartialTransaction#finish` method is in Engine where I have to handle explicitly all the 3 cases.

Reply:
That's fair. We are throwing untyped `new RuntimeException`s elsewhere though; I'd appreciate it if, somehow, some more meaningful types emerged for those.

Reply:
Those untyped `new RuntimeException`s are part of dam-script. I prefer not touching too much this part. 
cc @cocreature